paper-badge.html

```diff
@@ -116,6 +132,25 @@
         label: {
           type: String,
           observer: '_labelChanged'
+        },
+
+        /**
+         * An iron-icon ID. When given, the badge content will use an
+         * `<iron-icon>` element displaying the given icon ID rather than the
+         * label text. However, the label text will still be used for
+         * accessibility purposes.
+         */
+        icon: {
+          type: String,
+          value: ''
```

Comment:
If you remove the default value on this `icon` property, then you can remove your computed property below (because by default this will be undefined, so I think `hidden$="{{!icon}}"` will do the right thing.